app/controllers/profiles_controller.rb

```diff
@@ -64,7 +64,7 @@ def path_to_become_user
     end
 
     def path_to_unbecome_user
-      if current_user.admin?
+      if current_user.impersonator.admin?
```

Comment:
I believe with the added change to UserDecorator as above, this would read `current_user.as_deputy.admin?` which may or may not make more sense....